### DISCUSSION TITLE
App pod logs with linkerd | unable to view

### DISCUSSION THREAD

**Discussion by Chakki1301** [linkerd/linkerd2#9021] [Q&A] [answered]
I was able to view the app container logs using kubectl -f logs <pod> and was able to login to the container using "k exec --stdin --tty <POD> -- /bin/bash"
After injecting linkerd, I could not login to the container. However my goal is to check the app logs.
When I use this "k logs -f <POD> linkerd-proxy" I could not see the app-related logs. I tried injecting debug-sidecar as well. 
Tried this - "k logs deploy/<deploy> linkerd-debug -  " and as well as this "k exec -it <POD> -c linkerd-debug -- tshark -i any -f "tcp" -V -Y "http.request" 
still I couldn't see the exact logs for my app in the pod. Please suggest.

**Comment by mateiidavid** [comment, marked as answer]
I've answered a similar question in #9001: https://github.com/linkerd/linkerd2/issues/9001#issuecomment-1195244471

You will want to specify your container by name when interacting with a pod through `kubectl`. Assuming your pod has two containers: `linkerd-proxy` and `app`, you'd want to do `kubectl logs -f <pod> -c app` or `kubectl logs -f <pod> --all-containers=true` if you want the logs for all containers.

I'd suggest reading through `kubectl logs --help` for more info.
